lib/ansible/modules/network/haproxy.py

```diff
@@ -298,16 +321,30 @@ def wait_until_status(self, pxname, svname, status):
         the expected status in that time, the module will fail. If the service was
         not found, the module will fail.
         """
+        in_correct_state = False
         for i in range(1, self.wait_retries):
             state = self.get_state_for(pxname, svname)
 
             # We can assume there will only be 1 element in state because both svname and pxname are always set when we get here
             if state[0]['status'] == status:
-                return True
+                in_correct_state = True
+                if self.drain and status == 'MAINT':
+                    #we are in main, now check if the we are currently at 0 sessions
```

Comment:
Follow pep8 for block comments.

`Each line of a block comment starts with a # and a single space`